doc/developer-guide/internal-libraries/buffer-writer.en.rst

```diff
@@ -0,0 +1,302 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+   or more contributor license agreements.  See the NOTICE file
+   distributed with this work for additional information
+   regarding copyright ownership.  The ASF licenses this file
+   to you under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+
+.. include:: ../../common.defs
+
+.. highlight:: cpp
+.. default-domain:: cpp
+
+.. _BufferWriter:
+
+BufferWriter
+*************
+
+:class:`BufferWriter` is designed to make writing text to a buffer fast and safe. The output buffer
```

Comment:
class BufferWriter is designed to make writing text to a buffer convenient and safe.  It is faster and less error-prone than using sprintf().

Reply:
I'll update that on the next documentation pass, which I now have to do because @zwoop.